lib/schemaUtils.js

```diff
@@ -1386,27 +1427,38 @@ module.exports = {
   /**
    * Create parameters specific for a request
    * @param {*} localParams parameters array
+   * @param {object} options - a standard list of options that's globally passed around. Check options.js for more.
    * @returns {Object} with three arrays of query, header and path as keys.
    * @no-unit-test
    */
-  getParametersForPathItem: function(localParams) {
-    var tempParam,
+  getParametersForPathItem: function(localParams, options) {
+    let tempParam,
       params = {
         query: [],
         header: [],
         path: []
-      };
+      },
+      includeDeprecated = options.includeDeprecated !== undefined ?
+        options.includeDeprecated : true;
 
     _.forEach(localParams, (param) => {
       tempParam = param;
+      let shouldAddDeprecated = (includeDeprecated ||
+        includeDeprecated === false && tempParam.deprecated !== true);
       if (tempParam.in === 'query') {
-        params.query.push(tempParam);
+        if (shouldAddDeprecated) {
```

Comment:
Instead having same condition in each block, this can be put at start once which will take care of all if/else if blocks.

Reply:
Done